src/main/java/net/neoforged/camelot/commands/utility/ThreadPingsCommand.java

```diff
@@ -0,0 +1,261 @@
+        boolean isGuildId = channelId == event.getGuild().getIdLong();
+
+        final GuildChannel channel = event.getJDA().getGuildChannelById(channelId);
+        if (!isGuildId && channel == null) {
+            // TODO: handle this?
+            return;
+        }
+
+        final List<Role> roles = event.getInteraction().getMentions().getRoles();
+        final List<Long> roleIds = roles
+                .stream()
+                .filter(c -> c.getGuild().equals(event.getGuild())) // In case roles from other guilds are included
+                .map(ISnowflake::getIdLong)
+                .toList();
+
+        Database.pings().useExtension(ThreadPingsDAO.class, threadPings -> {
+            final List<Long> existingRoles = threadPings.query(channelId);
+
+            for (Long existingRoleId : existingRoles) {
+                if (!roleIds.contains(existingRoleId)) {
+                    threadPings.remove(channelId, existingRoleId);
+                }
+            }
+
+            for (Long roleId : roleIds) {
+                if (!existingRoles.contains(roleId)) {
+                    threadPings.add(channelId, roleId);
+                }
+            }
+        });
+
+        event.getInteraction().editMessage(buildMessage(isGuildId ? "this guild" : channel.getAsMention(), roles)).queue();
+    }
+
+    public static class ConfigureChannel extends SlashCommand {
+        public ConfigureChannel() {
+            this.name = "configure-channel";
+            this.help = "Configure roles to be pinged in threads made under a channel";
+            this.options = List.of(
+                    new OptionData(OptionType.CHANNEL, "channel", "The channel", true)
+            );
+        }
+
+        @Override
+        protected void execute(SlashCommandEvent event) {
+            final CommonResult result = executeCommon(event);
+            if (result == null) return;
+            result.interaction.getHook().editOriginal(buildMessage(result.channel.getAsMention(), result.roles))
+                    .setComponents(ActionRow.of(
+                            EntitySelectMenu.create(SELECT_MENU_ID_PREFIX + result.channel.getId(), SelectTarget.ROLE)
+                                    .setMinValues(0)
+                                    .setMaxValues(SelectMenu.OPTIONS_MAX_AMOUNT)
+                                    .build()
+                    ))
+                    .queue();
+        }
+    }
+
+    public static class ConfigureGuild extends SlashCommand {
+        private static final String SELECT_MENU_ID_PREFIX = "thread-pings-configure-";
+
+        public ConfigureGuild() {
+            this.name = "configure-guild";
+            this.help = "Configure roles to be pinged in threads made under this guild";
+        }
+
+        @Override
+        protected void execute(SlashCommandEvent event) {
+            event.getInteraction().deferReply(true).queue();
+            assert event.getGuild() != null;

```

Comment:
This also shouldn't be possible.

Reply:
Yes, but to appease IntelliJ IDEA's nullability analysis (later on, I use `event.getGuild()`) without adding a precondition, I add an assertion (which, as you know, are disabled at runtime unless enabled by VM flag). For this use-case, I consider it equivalent to e.g., `requireNonNull(event.getGuild()); // Should never throw`.